File: ios/MaterialsModule.swift
import ExpoModulesCore

public class MaterialsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("Materials")

    View(MaterialsView.self) {
      Prop("material") { (view: MaterialsView, material: String) in
        view.setMaterial(material)
      }
    }
  }
}

File: android/src/main/java/com/janblazej/materials/MaterialsModule.kt
package com.janblazej.materials

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class MaterialsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("Materials")

    View(MaterialView::class) {
      Prop("material") { view: MaterialView, material: String ->
        view.setMaterial(material)
      }

      Prop("variant") { view: MaterialView, variant: String? ->
        view.setVariant(variant)
      }
    }
  }
}
